holidays/countries/gibraltar.py

```diff
@@ -0,0 +1,220 @@
+        * [2020](https://web.archive.org/web/20250714135425/https://www.gibraltarlaws.gov.gi/uploads/legislations/banking-and-financial-dealings/B&P%20Holidays/2019s217.pdf)
+        * [2021](https://web.archive.org/web/20250421063152/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2020-5593/download)
+        * [2022](https://web.archive.org/web/20250414134257/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2021-6286/version/21-10-2021/download)
+        * [2023](https://web.archive.org/web/20240831014104/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2022-6735/version/22-02-2023/download)
+        * [2024](https://web.archive.org/web/20250502030403/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2023-7215/version/06-11-2023/download)
+        * [2025](https://web.archive.org/web/20250714135600/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2024-7456/download)
+    """
+
+    country = "GI"
+    default_language = "en_GB"
+    # %s (observed).
+    observed_label = "%s (observed)"
+    # First holiday info available from 2000.
+    start_year = 2000
+    supported_languages = ("en_GB", "en_US")
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        StaticHolidays.__init__(self, GibraltarStaticHolidays)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        self._add_observed(
+            # New Year's Day.
+            self._add_new_years_day(tr("New Year's Day")),
+        )
+
+        if self._year >= 2023:
+            # Winter Midterm Bank Holiday.
+            name = tr("Winter Midterm Bank Holiday")
+            if self._year == 2024:
+                self._add_holiday_2nd_mon_of_feb(name)
+            else:
+                self._add_holiday_3rd_mon_of_feb(name)
+
+        if self._year <= 2022:
+            # Commonwealth Day.
+            name = tr("Commonwealth Day")
+            if self._year >= 2021:
+                self._add_holiday_3rd_mon_of_feb(name)
+            else:
+                self._add_holiday_2nd_mon_of_mar(name)
+
+        # Good Friday.
+        self._add_good_friday(tr("Good Friday"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Easter Monday"))
+
+        # May Day.
+        name = tr("May Day")
+        special_dts = {
+            2007: (MAY, 7),
+            2008: (MAY, 5),
+            2009: (MAY, 4),
+            2010: (MAY, 3),
+            2011: (MAY, 2),
+            2013: (MAY, 6),
+            2016: (MAY, 2),
+            2021: (MAY, 3),
+        }
+        if dt := special_dts.get(self._year):
+            self._add_holiday(name, dt)
+        else:
+            self._add_observed(self._add_labor_day(name))
```

Comment:
```suggestion
        may_day_dts = {
            2007: (MAY, 7),
            2008: (MAY, 5),
            2009: (MAY, 4),
        }
        if dt := may_day_dts.get(self._year):
            self._add_holiday(name, dt)
        else:
            self._add_observed(self._add_holiday_may_1(name))
```
`_add_holiday_may_1` - because it looks like this holiday is not related to Labor Day.